src/main/java/org/jenkinsci/plugins/mesos/JenkinsScheduler.java

```diff
@@ -261,6 +266,19 @@ public void disconnected(SchedulerDriver driver) {
   public synchronized void resourceOffers(SchedulerDriver driver, List<Offer> offers) {
     LOGGER.fine("Received offers " + offers.size());
     for (Offer offer : offers) {
+      if (requests.isEmpty()) {
+        // decline offer for a longer period if no slave is waiting to get spawned
+        // this prevents getting offers every few seconds
```

Comment:
how about:

This prevents unnecessarily getting offers every few seconds and causing starvation when running a lot of frameworks.

You can kill the lines about marathon and chronos.